### PR TITLE
add self to path

### LINKED ISSUE #2282: Add Path::self()
Add an easy way to obtain the executable path. win/linux/mac only for now.
https://stackoverflow.com/a/1024937/205386

### PR DESCRIPTION
**Pull request by JackyWoo**
This pr try to close #2282

### CODE REVIEW

File: Foundation/testsuite/src/PathTest.cpp [hunk 1]
```diff
@@ -1619,6 +1619,28 @@ void PathTest::testWindowsSystem()
...
+	|| POCO_OS == POCO_OS_FREE_BSD                            \
+	|| POCO_OS == POCO_OS_SOLARIS                             \
+	|| POCO_OS == POCO_OS_LINUX                               \
+	|| POCO_OS == POCO_OS_ANDROID
+
+	assertTrue(!self.empty());
+	Path p(self);
+	assertTrue(p.isAbsolute());
+	assertTrue(p.isFile());
+	assertTrue(self.find("testrunner") != std::string::npos);
```
Comment (aleks-f) [inline Foundation/testsuite/src/PathTest.cpp:1637]: this test must somehow be skipped on windows because the executable is named `TestSuite` there

**Review by aleks-f — CHANGES_REQUESTED** [review]

**Comment by JackyWoo** [comment]
@aleks-f Pls review if you have a changce.

1 test error, maybe unrelated. 
```
There was 1 failure: 
 1: class CppUnit::TestCaller<class PollSetTest>.testPollClosedServer
    "fail: "
    in "<unknown>", line -1
```

**Comment by aleks-f** [comment]
> @aleks-f Pls review if you have a changce.

I [did review](https://github.com/pocoproject/poco/pull/3758/files#diff-579de81f02f258bbc25f9cd2945e7bf78660cc2ebfe6e510ae6d77a262a72061R48), but I do not se anything done about it

File: Foundation/testsuite/src/PathTest.cpp (hunk 1 above)
Comment (JackyWoo) [reply Foundation/testsuite/src/PathTest.cpp:1639]: @aleks-f I skipped and assertTrue on Windows. Or you mean skip `testSelf` on Windows?

**Comment by JackyWoo** [comment]
@aleks-f I wonder whether [Skip PathTest::testSelf on Windows](https://github.com/pocoproject/poco/pull/3758/commits/775afefa04660f89e9201a1f7b8122b64c811cc2) is what your meaning

File: Foundation/src/Path_UNIX.cpp [hunk 2]
```diff
@@ -32,6 +41,45 @@
 
 namespace Poco {
 
+std::string PathImpl::selfImpl()
+{
+	std::string path;
+#if POCO_OS == POCO_OS_MAC_OS_X
+	char buf[PATH_MAX];
```
[the remark this answers is not in the document]
Comment (aleks-f) [reply Foundation/src/Path_UNIX.cpp:48]: @JackyWoo do you plan to fix this? I was planning to release it in 1.13.0

**Review by aleks-f — CHANGES_REQUESTED** [review]

File: Foundation/testsuite/src/PathTest.cpp (hunk 1 above)
Comment (aleks-f) [reply Foundation/testsuite/src/PathTest.cpp:1638]: @JackyWoo windows should be tested

**Review by aleks-f — CHANGES_REQUESTED** [review]

File: Foundation/src/Path_UNIX.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackyWoo) [reply Foundation/src/Path_UNIX.cpp]: Yes,  fixed

**Comment by aleks-f** [comment]
#4319

**Review by aleks-f — COMMENTED** [review]
#4319

**Pull request closed without merging**